lib/double_entry/reporting/line_aggregate_filter.rb

```diff
@@ -0,0 +1,50 @@
+# encoding: utf-8
+module DoubleEntry
+  module Reporting
+    class LineAggregateFilter
+
+      def initialize(account, code, range, named_scopes)
+        @account = account
+        @code = code
+        @range = range
+        @named_scopes = named_scopes
+      end
+
+      def filter
+        @collection ||= apply_filters
+      end
+
+    private
+
+      attr_reader :account, :code, :range, :named_scopes
+
+      def apply_filters
+        collection = filter_collection.
+                     where(:account => account).
+                     where(:created_at => range.start..range.finish)
+        collection = collection.where(:code => code) if code
+
+        collection
+      end
+
+      # a lot of the trickier reports will use filters defined
+      # in named_scopes to bring in data from other tables.
+      def filter_collection
+        if named_scopes
+          collection = DoubleEntry::Line
+          named_scopes.each do |named_scope|
+            if named_scope.is_a?(Hash)
+              method_name = named_scope.keys[0]
+              collection = collection.send(method_name, named_scope[method_name])
```

Comment:
@orien check out https://gist.github.com/asellitt/25016cf728a418e57078#file-accounting_report_example-rb it is likely going to cause this to change significantly. Let me know what you think too!